lib/spawn-watcher.js

```diff
@@ -44,79 +40,41 @@ module.exports = {
     this.notify('Build success!', 'All Good!');
   },
 
-  clear (clearTerminal) {
-    this.clearTerminal = clearTerminal;
-    return this;
-  },
-
   handleError (err) {
+    this.errored = true;
+
     // log the error
-    const strErr = '' + err;
-    for (const line of strErr.split('\n')) {
-      log.error('s', red(line));
+    const strErr = `${err}`;
+    for (const line of strErr.split(EOL)) {
+      log.error('', red(line));
     }
 
     // use the system notifier
     const notifyErr = strErr.replace(COLOR_CODE_REGEXP, '');
     this.notify('Build failure!', notifyErr);
-    process.exit(1);
+    if (this.exitOnError) {
+      process.exit(1);
+    }
   },
 
   configure (taskName, filePattern, sequence) {
-    const subtaskName = `_${taskName}`;
-    const isRespawn = process.argv.includes('--respawn');
-
-    const watchFn = () => {
-      return new B((resolve) => {
-        (this.gulp.series(...sequence, function finishWatchSequence (done) {
-          resolve();
-          done();
-        }))();
-      });
+    const notifyWatch = (done) => {
+      if (!this.errored) {
+        this.notifyOK();
```

Comment:
should it also notify about errors?

Reply:
It does, in the tasks that produce errors (lint, test, whatever the consuming package has for watch tasks).